OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -30,6 +33,7 @@ public Attack(Actor self, Target target, WDist minRange, WDist maxRange, bool al
 			Target = target;
 			this.minRange = minRange;
 			this.maxRange = maxRange;
+			rangeModifiers = self.TraitsImplementing<IRangeModifier>();
```

Comment:
I'm not asking to cache the results (e.g. by calling ```ToArray``` or ```ToList```) - just the enumerable. Remember ```Select``` is lazy so the enumerable will get re-evaluated each time.